src/runtime/generators.js

```diff
@@ -175,6 +216,11 @@
     },
     throw: function(v) {
       return nextOrThrow(this[ctxName], this[moveNextName], 'throw', v);
+    },
+    return: function (v) {
+      this[ctxName].returnValue = v;
+      return nextOrThrow(this[ctxName], this[moveNextName], 'throw',
+        new StopIteration());
```

Comment:
This was an artefact from an earlier solution, where I stored the returnValue sent to the generator as a property of StopIteration.

I now made `new StopIteration()` a unique singleton object, named `SENTINEL`.